spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkWriteBuilder.java

```diff
@@ -117,7 +119,10 @@ public WriteBuilder overwrite(Filter[] filters) {
   @Override
   public Write build() {
     // Validate
-    Schema writeSchema = validateOrMergeWriteSchema(table, dsSchema, writeConf);
+    Schema writeSchema =
+        validateOrMergeWriteSchema(
+            table, dsSchema, writeConf, TableUtil.formatVersion(table) >= 3 && overwriteFiles);
```

Comment:
Add a comment for this, but in the context of building a regular `SparkWrite` (so appends, overwriteFIles, overwriteDynamicPartitions)  the only one case where we need to include row lineage explicitly in the output schema is the `overwriteFiles` case for CoW. 

In every other case, both fields will be explicitly null for all the fields produced, and as per the spec that means we do not need to write them out. cc @rdblue